File: Data Structures & Algorithms/depth-of-binary-tree/submission-0.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    void dfs(TreeNode* root,int count,int &max_count){
        if(!root)   return;
        if(root) count++;
        if(!root->left && !root->right) max_count=max(max_count,count);
        dfs(root->left,count,max_count);
        dfs(root->right,count,max_count);
    }
    int maxDepth(TreeNode* root) {
        int count=0;
        int max_count=0;
        dfs(root, count, max_count);
        return max_count;
    }
};

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        s=set(nums)
        maxCount=0
        for num in nums:
            if num-1 in s:
                continue
            count=1
            while num+1 in s:
                num=num+1
                count+=1
            maxCount=max(maxCount,count)
        return maxCount